### PR TITLE
Math magicians: components

### PR DESCRIPTION
**Pull request by zmekonnen251**
## Description

In this PR, I built a simple calculator UI which is going to be used in the next milestone. The UI looks like the image below.

calculator

### General requirements

- [x] Followed the **Github flow**.
- [x] There are no linter errors.
- [x] The documentation is professional.

### JavaScript requirements

- [x] Followed list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### Project requirements

- [x] created a React application is using **Create React App** (CRA) and following the instructions in this [guide](https://reactjs.org/docs/create-a-new-react-app.html#create-react-app).
- [x] The application runs in the browser without errors.
- [x]  Created a directory called `components`.
- [x] Inside `components`, created a new `Calculator.js` file.

Thank you for taking the time to review this pull request.

### CODE REVIEW

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @zmekonnen251,** 

### **Good job so far!**
You still need to work on some issues to go to the next project, but you are almost there!

## **Highlights**
- The PR is good👌.
- Linters are passing 👌.
- The readme file is good👌. 


## Required Changes ♻️
_Check the comments under the review._

- The Project components are really good but you have small details to fix, `App.js & Calculator.js` should be class-based.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js [hunk 1]
```diff
@@ -1,27 +1,10 @@
-import logo from './logo.svg';
 import './App.css';
+import Calculator from './components/Calculator';
 
 function App() {
```
Comment (Mihndim2020) [inline src/App.js]: - [x] As requested by the previous code reviewer, kindly change your `App.js` component to a `class-based` component as it is a requirement for this project.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
## Hi @zmekonnen251, 

Great work on making the changes requested by a previous reviewer 👏 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
stay-strong-youre-almost-there

 # Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by zmekonnen251** [comment]
- Dear Reviewer I merged by mistake and reverted back . Please consider it is my last chance, I did not merge it intentionally. Thank you for understanding.

**Review by julie-ify — COMMENTED** [review]
Hi @zmekonnen251, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by zmekonnen251** [comment]
Thank you @julie-ify for reviewing my project. :pray:

**Pull request merged**